hbase-server/src/main/java/org/apache/hadoop/hbase/replication/regionserver/ReplicationSource.java

```diff
@@ -587,12 +617,25 @@ private void initialize() {
 
   @Override
   public void startup() {
-    // mark we are running now
-    this.sourceRunning = true;
-    initThread = new Thread(this::initialize);
-    Threads.setDaemonThreadRunning(initThread,
-      Thread.currentThread().getName() + ".replicationSource," + this.queueId,
-      this::uncaughtException);
+    //Flag that signalizes uncaught error happening while starting up the source
+    // and a retry should be attempted
+    AtomicBoolean retryStartup = new AtomicBoolean(false);
```

Comment:
Why we need a AtomicBoolean here? It is only used locally, so a simple boolean is enough?

Reply:
It's been modified by the lambda expression on line #635, since lambdas expressions requires variables to be final, I can't use a simple, local primitive boolean or wrapper.

Reply:
Then use MutableBoolean? We do not need to use atomic here.

Reply:
Sure, thanks for suggesting MutableBoolean, am not too familiar with apache commons lib.